server/gateway.js

```diff
@@ -12,8 +12,13 @@ module.exports = function createGateway(collection) {
     },
     async create(newTodo) {
       const todo = Object.assign({ id: uuidv4() }, newTodo)
-      const created = await collection.insertOne(todo)
-      return created.result
+      try {
+        await collection.insertOne(todo)
+        return todo
+      }
+      catch (error) {
+        return error
```

Comment:
You can `omit` this `try/catch` block. If this method throws, it'll be caught in your `POST` handler.